### ISSUE TITLE
Option to create default config on first startup

### ISSUE THREAD

**Issue by SylvanFranklin** [Uzaaft/heimdall#3]
Yo this project looks sick and I think it could really take off with a little polish. I think on the first run it should either walk you through dialogue for creating a config file, or just go ahead and create an empty one for you so that there is no panic. I would love to contribute to this project if you are looking for help.

**Comment by Uzaaft** [comment]
We should perhaps think about the implementation before creating it.

An option I can think about is to prompt/notify the user if a config isn't found. 

Either trough the notifications system, or through stdout and then exit gracefully instead.

Creating an empty one might not be bueno since people don't like it when CLI tools add files from my experience.
Perhaps the default behavior can be added with a flag? I've been thinking about adding clap as a dependency anyways

**Comment by SylvanFranklin** [comment]
Okay I agree with the point about automatic creation. I think that the best approach would be just a std warning since the user is likely already in the terminal and it's just cleaner and more unix than a notification warning. 

A command line argument with clap is a solid idea. 

I can see it working like `heim --init`, which would scan for a config and put in a default one if none is detected. The init behavior could then also be easily added as a prompt option when the service is started for instance: 

```sh
starting...
No config file found at $XDG_CONFIG_HOME/heimdall/config.toml, would you like to create one? (Y/n)
```

**Comment by Uzaaft** [comment]
Agreed. But on macOS, you can install it trough brew, and use it as a service. For those, i think a notification would be nice since else, you don't really know if somebody failed unless you check brew.

**Comment by Uzaaft** [comment]
The service path is the main way I use heimdall. :)

**Comment by SylvanFranklin** [comment]
Okay your call, Aerospace does a similar thing and it work's pretty well.

**Comment by Uzaaft** [comment]
Let's go with the basic implementation without notifications first, and then we'll add in notifications at a later stage. Feel free to assign the issue to yourself if you want to have a go at it. Or else I can attack this next weekend.

**Comment by SylvanFranklin** [comment]
I can look at it during the next couple of days (: I think you have a very promising project here. It would be great to have more eyes on this... maybe make the README a little more official looking and post on Reddit? I think the name is also Sick, but there is a fairly large project by the same name also on github which is something to consider.

**Comment by Uzaaft** [comment]
> I can look at it during the next couple of days (: I think you have a very promising project here. It would be great to have more eyes on this... maybe make the README a little more official looking and post on Reddit? I think the name is also Sick, but there is a fairly large project by the same name also on github which is something to consider.

Haha I'm up to it if you think so. The project started as something I needed since I was tired of skhd.
And yeah, I know about that other project, but honestly I didn't care enough about the name collision 😂

**Comment by Uzaaft** [comment]
> I can look at it during the next couple of days (: I think you have a very promising project here. It would be great to have more eyes on this... maybe make the README a little more official looking and post on Reddit? I think the name is also Sick, but there is a fairly large project by the same name also on github which is something to consider.

What do you mean with "official looking"?

**Comment by SylvanFranklin** [comment]
Cool sounds good! I just mean there are some weird formatting things like the second number in the getting started list is inside the code block, and there are a bunch of same size title lines at the top which might look cleaner if broken down a bit. You could probably make the alpha warning and status into one smaller line with some emojis or "!!". Not a huge problem but I think keeping the warning small shows that it's ready for contribution and is more than a pile of spaghetti.

**Comment by Uzaaft** [comment]
> Cool sounds good! I just mean there are some weird formatting things like the second number in the getting started list is inside the code block, and there are a bunch of same size title lines at the top which might look cleaner if broken down a bit. You could probably make the alpha warning and status into one smaller line with some emojis or "!!". Not a huge problem but I think keeping the warning small shows that it's ready for contribution and is more than a pile of spaghetti.

I’ll cook up a cleaner readme this week. 🍝🍝

**Comment by Uzaaft** [comment]
Working on this.
